### ISSUE TITLE
Kramco prevents buying Torso

### ISSUE THREAD

**Issue by xande-1** [soolar/sl_ascend#163]
Tricky interaction issue here.  The logic for buying torso awaregness will only do so if we have at least 10k meat on hand.  But the logic for the Kramco spends us down to 5k meat on magical sausages whenever we have spare meat.  So an account with both a Kramco and a garbage tote will never accumulate enough meat  to buy torso and use the garbage shirt.  (Unless you do so manually, which I had been doing for other reasons, thus why I only discovered this interaction today.)

Probably the best solution is to add a conditional to the Kramco handling so the script will only spend down to 10k on magical sausages if you have the garbage tote && don't have torso && are a gnome sign.

Assuming I'm not missing something (which is always possible).

**Comment by xande-1** [comment]
Side note:  If the script does get to 10k meat before unlocking the beach, it spends 5k on a desert bus pass to avoid the Gnoll Garage.  (Apparently it already holds out for 10k meat if it might need a desert bus pass.)  For the accounts I run the script on, this (even without the Kramco interaction) would push torso to day 2 at the earliest.  I don't know if it's a common enough scenario to be worth worrying about, and I can't even say for sure that losing day 1 torso costs more turns than manually building a meatcar  (i.e., buying the bus pass may be optimal even if it delays torso), but I thought I'd mention it as something to ponder while that chunk of code is under review.

**Comment by jaspercb** [comment]
I agree that DG currently has a lot of meat contention. I tried pushing Kramco usage to only be after the first 150 turns, but perhaps that wasn't enough. Hm.

**Comment by xande-1** [comment]
For my (low-iotm, but having both a tote and Kramco) multis, who are on their 8th runs, as a gnome sign they have to get around 20k meat drops in order for torso to be bought (5k that the script keeps banked, 5k desert bus pass, 5k torso skill, 2k all-purpose flower, which I buy manually to make food when the script first runs out of turns on turn 40, 1.5k for 3 shore trips, plus clovers, and some other miscellaneous expenses).  While an account with all the shinies might pull that off, there's no way my multis are getting that much meat to drop before turn 150.  (PS - I manually buy Flesh Scent, so their meat drop isn't totally pathetic)

On my current run, the multis ended day 1 at about turn 152, and got up to about 6k meat on hand before they started grinding sausages (after hermit, all-purpose flower, and bus pass), so I just manually bought torso after the script finished, before logging out. (wiping out their meat on hand, but at this point they've got all their necessary early-game spending out of the way, so I don't think it will break anything.)

If my situation winds up being too niche, I can always just manually buy torso at the end of D1.  I wasn't aware of the 150 turns thing, so it may be doing fine for most people who have that combo of IotMs, since they also have others that help cover the meat drop issues.

PS - opening a separate issue for an NEP meat-drop thing I noticed, though.

**Comment by xande-1** [comment]
An alternate option is to let it buy torso with only 5k meat on hand as long as the beach is open and shore trips are done (and all-purpose flower is bought, possibly).  There's not a lot of meat contention once those things are out of the way for several levels, so depleting the meat on hand shouldn't be a big issue (at least it hasn't been for me when I've done it).
